pkg/alexa/gen/skill.go

```diff
@@ -6,415 +6,283 @@ import (
-		// ensure that a default is set
-		if s.defaultLocale == "" {
-			s.defaultLocale = l
-		}
+func (s *SkillBuilder) WithLocaleTestingInstructions(locale string, instructions string) *SkillBuilder {
+	loc, err := s.registry.Resolve(locale)
+	if err != nil {
+		return s
 	}
-	s.locales[l] = LocaleDef{Translations: trans}
+	loc.Set(s.skillInstructions, []string{instructions})
+	return s
 }
 
-func (s *SkillBuilder) AddIntent(intent Intent) {
-	s.intents = append(s.intents, intent)
+func (s *SkillBuilder) WithPrivacyFlag(flag string, value bool) *SkillBuilder {
+	s.privacyFlags[flag] = value
+	return s
 }
 
-// AddIntentString creates an Intent from the string, adds it and returns a reference.
-func (s *SkillBuilder) AddIntentString(intent string) *Intent {
-	i := NewIntent(intent)
-	s.intents = append(s.intents, i)
-	return &i
-}
+func (s *SkillBuilder) AddLocale(locale string) *SkillLocaleBuilder {
+	s.registry.Register(l10n.NewLocale(locale))
 
-func (s *SkillBuilder) AddType(t Type) {
-	// TODO: validate rules!
-	s.types = append(s.types, t)
+	lb := NewSkillLocaleBuilder(locale).
+		WithLocaleRegistry(s.registry)
+	s.locales[locale] = lb
+	return lb
 }
 
-func (s *SkillBuilder) AddTypeString(t string) {
-	s.types = append(s.types, Type(t))
-}
-
-func (s *SkillBuilder) AddCountry(c alexa.Country) {
-	s.countries = append(s.countries, c)
+func (s *SkillBuilder) AddCountry(country string) *SkillBuilder {
+	s.countries = append(s.countries, country)
+	return s
 }
 
-func (s *SkillBuilder) AddCountries(cs []alexa.Country) {
+func (s *SkillBuilder) AddCountries(cs []string) *SkillBuilder {
 	for _, c := range cs {
 		s.countries = append(s.countries, c)
 	}
+	return s
 }
 
 // Build builds an alexa.Skill object.
-// TODO: return errors!
 func (s *SkillBuilder) Build() (*alexa.Skill, error) {
+	if s.registry == nil || len(s.registry.GetLocales()) == 0 {
+		return nil, fmt.Errorf("No locales registered to build")
+	}
+	//
+	if s.locales == nil || len(s.locales) == 0 {
+		for n, _ := range s.registry.GetLocales() {
+			s.locales[n] = NewSkillLocaleBuilder(n).
+				WithLocaleRegistry(s.registry)
+
+		}
+	}
+
+	dl := s.registry.GetDefault()
+
 	skill := &alexa.Skill{
 		Manifest: alexa.Manifest{
-			Version: "1.0",
+			Version:    "1.0",
+			Publishing: alexa.Publishing{},
 		},
 	}
-
-	// default is always set if at least one locale was defined.
-	dl, _ := s.locales[s.defaultLocale]
-
 	skill.Manifest.Publishing.Category = s.category
 	// TODO: ensure unique occurance
-	skill.Manifest.Publishing.Countries = s.countries
-
-	if s.testingInstructions != "" {
-		skill.Manifest.Publishing.TestingInstructions = s.testingInstructions
+	if len(s.countries) > 0 {
+		skill.Manifest.Publishing.Countries = s.countries
 	} else {
-		skill.Manifest.Publishing.TestingInstructions = dl.Translations.Get(l10n.KeySkillTestingInstructions)
+		skill.Manifest.Publishing.Worldwide = true
 	}
+	skill.Manifest.Publishing.TestingInstructions = dl.Get(s.skillInstructions)
```

Comment:
This seems unexpected and magical.

Reply:
Yes, it also was for me :) The Skill JSON only has one field for testing instructions (not per locale), so they are fetched from the default locale.